metatensor-torch/tests/labels.cpp

```diff
@@ -41,6 +41,24 @@ TEST_CASE("Labels") {
         CHECK_THROWS_WITH(LabelsHolder::create({}, {{}, {}, {}}), "invalid parameter: can not have labels.count > 0 if labels.size is 0");
     }
 
+    SECTION("unchecked constructor") {
+        auto names = std::vector<std::string>{"a", "b"};
+
+        auto non_unique_values = torch::tensor({{1, 2}, {1, 2}}, torch::kInt32);
+        CHECK_THROWS_WITH(
+            LabelsHolder(names, non_unique_values),
+            "invalid parameter: can not have the same label entry multiple times: [1, 2] is already present"
+        );
+        auto labels = LabelsHolder(names, non_unique_values, metatensor::unchecked_t{});
+        CHECK(labels.count() == 2);
+        CHECK(labels.values()[0][0].item<int64_t>() == 1);
+        CHECK(labels.values()[1][0].item<int64_t>() == 1);
+        CHECK(labels.values()[0][1].item<int64_t>() == 2);
+        CHECK(labels.values()[1][1].item<int64_t>() == 2);
+        auto created = LabelsHolder::create(names, {{1, 2}, {1, 2}}, metatensor::unchecked_t{});
+        CHECK(created->count() == 2);
```

Comment:
I don't like this: it should still be invalid to create Labels with the same entry twice, we are only giving the user the option to bypass slow checks, not to create invalid data.

Reply:
Aren't they one and the same? If we don't do the check we don't know if the data is valid right?